icu4c/source/test/intltest/tmsgfmt.cpp

```diff
@@ -2029,4 +2032,60 @@ void TestMessageFormat::TestMessageFormatNumberSkeleton() {
     }
 }
 
+void TestMessageFormat::doTheRealDateTimeSkeletonTesting(
+        UDate testDate, const char16_t* cases[][3], size_t caseCount, IcuTestErrorCode& status) {
+
+    for (size_t i = 0; i < caseCount; i++) {
+        auto& cas = cases[i];
+
+        const char16_t* messagePattern = cas[0];
+        const char16_t* expected = cas[2];
+        std::string locale;
+        UnicodeString ulocale(cas[1]);
+        ulocale.toUTF8String(locale);
```

Comment:
This "only" works if the platform is ASCII-based. You need to use a UnicodeString conversion function for "invariant characters" to get the locale ID in platform encoding.

Better:

A simpler way to do this would be to have this function take three parameters for the pattern, locale, and expected -- with Unicode vs. platform string types as appropriate -- rather than take the cases array, and just call the function multiple times rather than in a loop. Makes the code more readable, too, and easier to debug.

Reply:
Done